src/GeositeFramework/js/Plugin.js

```diff
@@ -47,7 +47,43 @@ require(['use!Geosite',
                         downloadAsCsv: requestCsvDownload,
                         downloadAsPlainText: requestTextDownload,
                         dispatcher: N.app.dispatcher,
-                        suppressHelpOnStartup: _.partial(suppressHelpOnStartup, model)
+                        suppressHelpOnStartup: _.partial(suppressHelpOnStartup, model),
+                        resize: {
+                            adjustBy: function(val) {
+                                var currentWidth = parseFloat($uiContainer.css("width"));
+                                var d = $.Deferred();
+                                d.then(function() {
+                                    $uiContainer.css({ width: currentWidth + val });
+                                });
+                                return d;
+                            },
+                            setWidth: function(val) {
+                                var d = $.Deferred();
+                                if (typeof (val) === "string") {
+                                    switch (val.toLowerCase()) {
+                                        case 'large':
+                                            d.then(function() {
+                                                $uiContainer.removeClass("sidebar-width-small");
+                                                $uiContainer.addClass("sidebar-width-large");
+                                            });
+                                            break;
+                                        case 'small':
+                                            d.then(function() {
+                                                $uiContainer.addClass("sidebar-width-small");
+                                                $uiContainer.removeClass("sidebar-width-large");
+                                            });
+                                            break;
+                                        default:
+                                            break;
+                                    }
+                                } else if (typeof (val) === "number") {
+                                    d.then(function() {
+                                        $uiContainer.css({ width: val });
```

Comment:
If the plugin ever sets this, then subsequently settings `small` or `large` will never work since the `element.style` will always override a class.  The safest approach would be to remove the `width` css anytime a class is set.  That in turn might push the `case` for both `large` and `small` do be DRY'd up into a function.